kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/KeyValueHeaderRecordWriterProvider.java

```diff
@@ -75,64 +75,86 @@ public RecordWriter getRecordWriter(S3SinkConnectorConfig conf, String filename)
         ? filename.substring(0, filename.length() - valueProvider.getExtension().length())
         : filename;
 
-    RecordWriter valueWriter = valueProvider.getRecordWriter(conf, strippedFilename);
-    RecordWriter keyWriter =
-        keyProvider == null ? null : keyProvider.getRecordWriter(conf, strippedFilename);
-    RecordWriter headerWriter =
-        headerProvider == null ? null : headerProvider.getRecordWriter(conf, strippedFilename);
+    Optional<RecordWriter> valueWriter = filename.contains(conf.getTombstoneEncodedPartition())
```

Comment:
if the output partition contains the tombstone partition pattern, then we determine this is for tombstone records, the ValueWriter is not needed